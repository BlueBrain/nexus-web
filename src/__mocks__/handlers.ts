import { rest } from 'msw';

export const handlers = [
  rest.get(
    'https://localhost:3000/resources/org/project/_/:Id',
    (req, res, ctx) => {
      const { Id } = req.params;
      const mockResponse = {
        '@context': [
          'https://bluebrain.github.io/nexus/contexts/metadata.json',
          'https://bluebrainnexus.io/studio/context',
        ],
        '@id': Id,
        '@type': 'Studio',
        description: `test description for ${Id}`,
        label: `label for ${Id}`,
        plugins: [
          {
            customise: true,
            plugins: [
              { expanded: false, key: 'video' },
              { expanded: false, key: 'preview' },
              { expanded: false, key: 'admin' },
              { expanded: false, key: 'circuit' },
              { expanded: false, key: 'image-collection-viewer' },
              { expanded: false, key: 'jira' },
              { expanded: false, key: 'markdown' },
            ],
          },
        ],
        workspaces: ['id-1', 'id-2', 'id-3', 'id-4'],
      };
      return res(
        // Respond with a 200 status code
        ctx.status(200),
        ctx.json(mockResponse)
      );
    }
  ),
  rest.get('https://localhost:3000/resources/org/project', (req, res, ctx) => {
    const mockResponse = {
      '@context': [
        'https://bluebrain.github.io/nexus/contexts/metadata.json',
        'https://bluebrain.github.io/nexus/contexts/search.json',
        'https://bluebrain.github.io/nexus/contexts/search-metadata.json',
      ],
      _total: 19,
      _results: [
        {
          '@id': 'id-1',
          label: 'test-label-1',
        },
        {
          '@id': 'id-2',
          label: 'test-label-2',
        },
        {
          '@id': 'id-3',
          label: 'test-label-3',
        },
        {
          '@id': 'id-4',
          label: 'test-label-4',
        },
        {
          '@id': 'id-5',
          label: 'test-label-5',
        },
        {
          '@id': 'id-6',
          label: 'test-label-6',
        },
        {
          '@id': 'id-7',
          label: 'test-label-7',
        },
        {
          '@id': 'id-8',
          label: 'test-label-8',
        },
        {
          '@id': 'id-9',
          label: 'test-label-9',
        },
        {
          '@id': 'id-10',
          label: 'test-label-10',
        },
      ],
    };

    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json(mockResponse)
    );
  }),
  rest.get('https://localhost:3000/resources', (req, res, ctx) => {
    const mockResponse = {
      '@context': [
        'https://bluebrain.github.io/nexus/contexts/metadata.json',
        'https://bluebrain.github.io/nexus/contexts/search.json',
        'https://bluebrain.github.io/nexus/contexts/search-metadata.json',
      ],
      _total: 19,
      _results: [
        {
          '@id': 'id-1',
          label: 'test-label-1',
          projectLabel: 'project',
          _project: '/org1/project1',
        },
        {
          '@id': 'id-2',
          label: 'test-label-2',
          _project: '/org/project',
        },
        {
          '@id': 'id-3',
          label: 'test-label-3',
          _project: '/org/project',
        },
        {
          '@id': 'id-4',
          label: 'test-label-4',
          _project: '/org/project',
        },
        {
          '@id': 'id-5',
          label: 'test-label-5',
          _project: '/org/project',
        },
        {
          '@id': 'id-6',
          label: 'test-label-6',
          _project: '/org/project',
        },
        {
          '@id': 'id-7',
          label: 'test-label-7',
          _project: '/org/project',
        },
        {
          '@id': 'id-8',
          label: 'test-label-8',
          _project: '/org/project',
        },
        {
          '@id': 'id-9',
          label: 'test-label-9',
          _project: '/org/project',
        },
        {
          '@id': 'id-10',
          label: 'test-label-10',
          _project: '/org/project',
        },
      ],
    };
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json(mockResponse)
    );
  }),
];
