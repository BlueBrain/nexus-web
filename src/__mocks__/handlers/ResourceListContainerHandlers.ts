import { rest } from 'msw';
import { deltaPath } from '__mocks__/handlers/handlers';
import { Resource } from '@bbp/nexus-sdk';

export const resourcesHandler = rest.get(
  deltaPath(`resources/bbp/agents`),
  (_, res, ctx) => {
    const mockResponse = {
      '@context': ['https://bluebrain.github.io/nexus/contexts/metadata.json'],
      _total: 3,
      _results: [
        getMockResource('1'),
        getMockResource('2'),
        getMockResource('3'),
      ],
    };

    return res(ctx.status(200), ctx.json(mockResponse));
  }
);

export const searchHitsHandler = rest.post(
  deltaPath(
    '/views/bbp/agents/https%3A%2F%2Fbluebrain.github.io%2Fnexus%2Fvocabulary%2FdefaultElasticSearchIndex/_search'
  ),
  (_, res, ctx) => {
    const filteredByDeprecation = {
      buckets: [getMockBucket('1'), getMockBucket('2'), getMockBucket('3')],
      doc_count_error_upper_bound: 0,
      sum_other_doc_count: 0,
    };
    const mockResponse = {
      aggregations: {
        schemas: {
          doc_count: 3,
          filteredByDeprecation: { ...filteredByDeprecation },
        },
        types: {
          doc_count: 8,
          filteredByDeprecation: { ...filteredByDeprecation },
        },
      },
      hits: {
        hits: [
          getMockSearchHit('1'),
          getMockSearchHit('2'),
          getMockSearchHit('3'),
        ],
        max_score: 123,
        total: { relation: 'eq', value: 11 },
      },
      timed_out: false,
      took: 0,
      _shards: { failed: 0, skipped: 0, successful: 1, total: 1 },
    };

    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json(mockResponse)
    );
  }
);

const getMockResource = (id: string, extra?: Partial<Resource>) => ({
  '@id': id,
  '@type': ['View'],
  description: 'Test description',
  name: 'Test name',
  _constrainedBy: 'https://bluebrain.github.io/nexus/schemas/views.json',
  _createdAt: '2024-01-19T11:40:24.804553Z',
  _createdBy: 'https://bbp.epfl.ch/nexus/v1/realms/bbp/users/mockuser',
  _deprecated: false,
  _incoming: 'test',
  _outgoing: 'test',
  _project: 'https://dev.nise.bbp.epfl.ch/nexus/v1/projects/bbp/agents',
  _rev: 1,
  _self: id,
  _updatedAt: '2024-01-19T11:40:24.804553Z',
  _updatedBy: 'https://bbp.epfl.ch/nexus/v1/realms/bbp/users/mockuser',
  _uuid: id,
  ...extra,
});

const getMockSearchHit = (id: string, extra?: Partial<Resource>) => {
  const resource = getMockResource(id, extra);
  return {
    _id: id,
    _index: `delta_${id}`,
    _score: 123,
    _source: {
      ...resource,
      _original_source: JSON.stringify(resource),
    },
    _type: '_doc',
  };
};

const getMockBucket = (key: string, docCount: number = 1) => ({
  key,
  doc_count: docCount,
});
