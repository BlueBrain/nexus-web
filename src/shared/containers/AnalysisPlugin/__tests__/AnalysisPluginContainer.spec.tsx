import { vi } from 'vitest';
import { NexusProvider } from '@bbp/react-nexus';
import { createNexusClient } from '@bbp/nexus-sdk/es';
import * as React from 'react';
import fetch from 'node-fetch';
import { QueryClient, QueryClientProvider } from 'react-query';
import AnalysisPluginContainer from '../AnalysisPluginContainer';
import { deltaPath } from '__mocks__/handlers/handlers';
import {
  render,
  server,
  waitFor,
  screen,
  cleanup,
} from '../../../../utils/testUtil';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import userEvent from '@testing-library/user-event';
import {
  sparqlAnalysisReportNoResultsHandler,
  resourcesAnalysisReportType,
  sparqlAnalysisReportSingleResult,
  imageResourceFile,
  reportResource,
} from '__mocks__/handlers/AnalysisPlugin/handlers';
import {
  DEFAULT_REPORT_CATEGORIES,
  DEFAULT_REPORT_TYPES,
} from '../../../../constants';

describe('Analysis Plugin', () => {
  const mockState = {
    config: {
      apiEndpoint: deltaPath(),
      analysisPluginSparqlDataQuery: 'detailedCircuit',
      analysisPluginTypes: DEFAULT_REPORT_TYPES,
      analysisPluginCategories: DEFAULT_REPORT_CATEGORIES,
    },
    auth: {
      identities: [],
    },
  };
  const queryClient = new QueryClient();
  const mockStore = configureStore();
  jest.mock('react-redux', () => {
    const ActualReactRedux = jest.requireActual('react-redux');
    return {
      ...ActualReactRedux,
      useSelector: vi.fn().mockImplementation(() => {
        return mockState;
      }),
    };
  });

  // establish API mocking before all tests
  beforeAll(() => server.listen());
  // reset any request handlers that are declared as a part of our tests
  // (i.e. for testing one-time error scenarios)
  afterEach(() => {
    cleanup();
    server.resetHandlers();
    queryClient.clear();
  });
  // clean up once the tests are done
  afterAll(() => server.close());
  const nexus = createNexusClient({
    fetch,
    uri: deltaPath(),
  });

  it('add new Analysis Report button is present', async () => {
    server.use(sparqlAnalysisReportNoResultsHandler);

    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      const addButton = screen.getByRole('button', {
        name: 'Add Report',
      });
      expect(addButton).toBeInTheDocument();
    });
  });

  it('clicking add New Analysis Report button results in screen displaying all required options to create new Analysis Report', async () => {
    server.use(sparqlAnalysisReportNoResultsHandler);
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      screen.getByRole('button', {
        name: 'Add Report',
      });
    });

    const addButton = await screen.findByRole('button', {
      name: 'Add Report',
    });
    user.click(addButton);

    expect(
      await waitFor(() => screen.getByRole('button', { name: 'Save' }))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByRole('button', { name: 'Cancel' }))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByRole('textbox', { name: 'Report Name' }))
    ).toBeInTheDocument();

    expect(
      await waitFor(() =>
        screen.getByRole('textbox', { name: 'Report Description' })
      )
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByText('3. Categories'))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByText('4. Types'))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByText('5. Add Assets'))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByText('6. Tools'))
    ).toBeInTheDocument();

    expect(
      await waitFor(() => screen.getByRole('button', { name: 'Add tool' }))
    ).toBeInTheDocument();
  });

  it('On Create New Analysis screen, clicking cancel will return to the view mode', async () => {
    server.use(sparqlAnalysisReportNoResultsHandler);
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      screen.getByRole('button', {
        name: 'Add Report',
      });
    });

    const addButton = await screen.findByRole('button', {
      name: 'Add Report',
    });
    user.click(addButton);

    const cancelBtn = await waitFor(() =>
      screen.getByRole('button', { name: 'Cancel' })
    );
    user.click(cancelBtn);

    expect(
      await waitFor(() =>
        screen.getByRole('button', {
          name: 'Add Report',
        })
      )
    ).toBeInTheDocument();
  });

  it('On Create New Analysis screen, clicking save will trigger analysis report to be saved', async () => {
    server.use(
      sparqlAnalysisReportNoResultsHandler,
      resourcesAnalysisReportType
    );
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() =>
      screen.getByRole('button', {
        name: 'Add Report',
      })
    );

    const addButton = await screen.findByRole('button', {
      name: 'Add Report',
    });
    user.click(addButton);

    const analysisNameTextBox = await waitFor(() =>
      screen.getByRole('textbox', { name: 'Report Name' })
    );

    user.type(analysisNameTextBox, 'New Report Name');

    const analysisDescriptionTextBox = await waitFor(() =>
      screen.getByRole('textbox', { name: 'Report Description' })
    );

    user.type(analysisDescriptionTextBox, 'New Report Description');

    const saveBtn = await waitFor(() =>
      screen.getByRole('button', { name: 'Save' })
    );

    user.click(saveBtn);
  });

  it('On Create New Analysis screen, clicking cancel will return to the view mode', async () => {
    server.use(sparqlAnalysisReportNoResultsHandler);
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      screen.getByRole('button', {
        name: 'Add Report',
      });
    });

    const addButton = await screen.findByRole('button', {
      name: 'Add Report',
    });
    user.click(addButton);

    const cancelBtn = await waitFor(() =>
      screen.getByRole('button', { name: 'Cancel' })
    );
    user.click(cancelBtn);

    await waitFor(() => {
      screen.getByRole('button', {
        name: 'Add Report',
      });
    });
  });

  it('On an individual analysis report, the option to navigate to the parent container resource is presented', async () => {
    server.use(
      sparqlAnalysisReportSingleResult,
      reportResource,
      imageResourceFile
    );
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysisReport1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });
    await waitFor(() => {
      screen.getByRole('button', {
        name: 'Go to parent resource',
      });
    });
  });

  it('On a container analysis resource, each individual analysis has an option to navigate to the resource', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysis1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });
    expect(
      await waitFor(() =>
        screen.getByRole('button', {
          name: /Open discussion on report resource/,
        })
      )
    ).toBeInTheDocument();
  });

  it('on initial load the first analysis report is visible', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    expect(
      await waitFor(() => screen.getByText('Our Very First Analysis Report!'))
    ).toBeInTheDocument();
  });

  it('when no analysis report selected, zoom options are hidden', async () => {
    server.use(sparqlAnalysisReportNoResultsHandler);
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="resourceId"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    expect(
      screen.queryByLabelText(/Increase\/Decrease/)
    ).not.toBeInTheDocument();
  });

  it('analysis report assets show image name (or filename if not present) along with last updated details', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    const user = userEvent.setup();
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysis1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );
    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });
    // expect asset name to be present
    expect(
      await waitFor(() => screen.getByText('insta_logo_large.png'))
    ).toBeVisible();
  });

  it('clicking analysis report asset opens preview of asset', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const user = userEvent.setup();
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysis1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    // expect asset name to be present
    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });
    const asset = await waitFor(() =>
      screen.getByLabelText(/insta_logo_large/)
    );
    user.click(asset);
    expect(
      await waitFor(() => screen.getByText('insta_logo_large.png'))
    ).toBeInTheDocument();
  });
  it('In edit mode, image delete button is visible', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const user = userEvent.setup({ pointerEventsCheck: 0 });
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysis1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });

    await waitFor(() => {
      const edit = screen.getByRole('button', { name: 'Edit Report' });
      user.click(edit);
    });

    const analysisFile = await waitFor(
      () => screen.getAllByLabelText('Report File')[0]
    );
    user.click(analysisFile);

    expect(
      await waitFor(() => screen.getByRole('button', { name: 'Delete' }))
    ).toBeInTheDocument();
  });

  it('In edit mode, user can add a new tool to the report', async () => {
    server.use(sparqlAnalysisReportSingleResult, reportResource);
    const user = userEvent.setup({ pointerEventsCheck: 0 });
    const history = createMemoryHistory({});
    const store = mockStore(mockState);
    render(
      <Router history={history}>
        <Provider store={store}>
          <QueryClientProvider client={queryClient}>
            <NexusProvider nexusClient={nexus}>
              <AnalysisPluginContainer
                projectLabel="projectLabel"
                orgLabel="orgLabel"
                resourceId="https://dev.nise.bbp.epfl.ch/nexus/v1/resources/bbp-users/nicholas/_/MyTestAnalysis1"
              ></AnalysisPluginContainer>
            </NexusProvider>
          </QueryClientProvider>
        </Provider>
      </Router>
    );

    await waitFor(() => {
      const title = screen.getByText('Our Very First Analysis Report!');
      user.click(title);
    });

    await waitFor(() => {
      const edit = screen.getByRole('button', { name: 'Edit Report' });
      user.click(edit);
    });

    await waitFor(() => {
      const addTool = screen.getByRole('button', { name: 'Add tool' });
      user.click(addTool);
    });

    expect(
      await waitFor(() => screen.getByText('Script Location'))
    ).toBeInTheDocument();
    expect(
      await waitFor(() => screen.getByText('How did you run the script?'))
    ).toBeInTheDocument();
  });
});
