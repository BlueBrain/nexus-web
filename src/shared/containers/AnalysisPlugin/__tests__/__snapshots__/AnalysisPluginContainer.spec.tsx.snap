// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analysis Plugin renders without data 1`] = `
<div>
  <div
    class="analysis"
  >
    <div
      class="analysisTools"
    >
      <div
        class="ant-select select-analysis ant-select-multiple ant-select-show-search"
        style="width: 100%;"
      >
        <div
          class="ant-select-selector"
        >
          <div
            class="ant-select-selection-overflow"
          >
            <div
              class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
              style="opacity: 1;"
            >
              <div
                class="ant-select-selection-search"
                style="width: 0px;"
              >
                <input
                  aria-activedescendant="undefined_list_0"
                  aria-autocomplete="list"
                  aria-controls="undefined_list"
                  aria-haspopup="listbox"
                  aria-owns="undefined_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
                <span
                  aria-hidden="true"
                  class="ant-select-selection-search-mirror"
                >
                  
                  Â 
                </span>
              </div>
            </div>
          </div>
          <span
            class="ant-select-selection-placeholder"
          >
            Select Analysis
          </span>
        </div>
      </div>
      <button
        aria-label="Add Analysis Report"
        class="ant-btn ant-btn-primary ant-btn-circle ant-btn-icon-only"
        title="Add Analysis Report"
        type="button"
      >
        <span
          aria-label="plus"
          class="anticon anticon-plus"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="plus"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <defs>
              <style />
            </defs>
            <path
              d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
            />
            <path
              d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      aria-label="Increase/Decrease image thumnbail size"
      class="zoom-control"
    >
      <span
        aria-label="zoom-out"
        class="anticon anticon-zoom-out"
        role="img"
        title="Reduce thumbnail size"
      >
        <svg
          aria-hidden="true"
          data-icon="zoom-out"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M637 443H325c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h312c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zm284 424L775 721c122.1-148.9 113.6-369.5-26-509-148-148.1-388.4-148.1-537 0-148.1 148.6-148.1 389 0 537 139.5 139.6 360.1 148.1 509 26l146 146c3.2 2.8 8.3 2.8 11 0l43-43c2.8-2.7 2.8-7.8 0-11zM696 696c-118.8 118.7-311.2 118.7-430 0-118.7-118.8-118.7-311.2 0-430 118.8-118.7 311.2-118.7 430 0 118.7 118.8 118.7 311.2 0 430z"
          />
        </svg>
      </span>
      <div
        class="ant-slider slider-scale"
      >
        <div
          class="ant-slider-rail"
        />
        <div
          class="ant-slider-step"
        />
        <div
          aria-disabled="false"
          aria-valuemax="100"
          aria-valuemin="0"
          aria-valuenow="50"
          class="ant-slider-handle"
          role="slider"
          style="left: 50%; transform: translateX(-50%);"
          tabindex="0"
        />
        <div
          class="ant-slider-mark"
        />
      </div>
      <span
        aria-label="zoom-in"
        class="anticon anticon-zoom-in"
        role="img"
        title="Increase thumbnail size"
      >
        <svg
          aria-hidden="true"
          data-icon="zoom-in"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M637 443H519V309c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v134H325c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h118v134c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V519h118c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8zm284 424L775 721c122.1-148.9 113.6-369.5-26-509-148-148.1-388.4-148.1-537 0-148.1 148.6-148.1 389 0 537 139.5 139.6 360.1 148.1 509 26l146 146c3.2 2.8 8.3 2.8 11 0l43-43c2.8-2.7 2.8-7.8 0-11zM696 696c-118.8 118.7-311.2 118.7-430 0-118.7-118.8-118.7-311.2 0-430 118.8-118.7 311.2-118.7 430 0 118.7 118.8 118.7 311.2 0 430z"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
